enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,435 @@
+
+Since indexes are manipulated by making additive changes to it, the way an index can we reproduced today is my managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) and by mananging a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, with every opportunity to miss documenting the history of an index and thereby loosing the ability to recreate an index. 
+
+### Goals
+
+A human consumable representation of an index in the form of json/yaml will: 
+
+1. Allow for validation of the structure of an index against a standard definition of an index.
+2. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+3. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+4. Allow for reproducability of an index using just the yaml/json representation of the index.   
+
+### Non-Goals
+
+1. Validation of the content of the index, i.e individual bundles. That is considered in the enhancement for [operator bundle validation](https://github.com/operator-framework/enhancements/pull/43).  
+
+### User Stories 
+
+#### Story 1
+
+As an index author, I can query an index for a json/yaml representation of the contents of an index. 
+
+For example, if an index `docker.io/my-namespace/my-index` has one bundle `quay.io/etcd` added to it,
+
+`opm represent --output=json docker.io/my-namespace/my-index` will output the following json:
+
+```output.json
+{
+    "graph": {
+        "nodes": [
+            {
+                "bundlePath": "quay.io/etcd",
+                "provides": [
+                    {
+                        "type": "olm.package",
+                        "version": "1.0.0",
+                        "packageName": "etcd"
+                    },
+                    {
+                        "type": "olm.gvk",
+                        "version": "v1beta2",
+                        "group": "etcd.database.coreos.com",
+                        "kind": "EtcdCluster"
+                    },
+                    {
+                        "type": "label",
+                        "value": "alpha"
+                    }
+                ],
+                "constraints": [
+                    [
+                        {
+                            "type": "label",
+                            "value": "LTS"
+                        }
+                    ]
+                ]
+            },
+        ]
+    }
+}
```

Comment:
can you update this example to illustrate an upgrade graph with multiple entries instead of just a singe operator version so it's clear that is what it is intended to represent (assuming that is correct)

Reply:
@bparees updated.